### DISCUSSION TITLE
Web Radio Replace & Play Not Working After 20230917 Update

### DISCUSSION THREAD

**Discussion by R101G** [rern/rAudio#1422] [Bug Reports]
@rern 
This is a regression. After selecting a radio station it does not play, or does not find the metadata, or plays for a second and stops.

rAudio 20230917
6.1.52-1 aarch64
Raspberry Pi 3B Rev 1.2
BCM2837 • 957 MiB
4 x Cortex-A53 @ 1.20 GHz

**Comment by klassikmann** [comment]
Can't confirm but something similar:
"Add + Play" or "Replace + Play" of a webradio often work like "Add" or "Replace", i.e. without immediate play. I have to click "Play" in the playback window.

**Reply by R101G** [reply]
I see this too, but none of the problems are consistent.

**Comment by ceejayemm** [comment]
Seems to work OK for me after applying 202030918 update - UK stations only.

Chris

**Comment by R101G** [comment]
Looks OK to me too. Thanks @rern.

**Comment by R101G** [comment]
@rern 
Sorry, I spoke too soon. The problem is still there when switching between Radio Paradise streams.

**Reply by rern** [reply]
@R101G, 

Please try the UPDATE branch.

**Reply by klassikmann** [reply]
First impression: Looking good

**Reply by R101G** [reply]
@rern 

Yes, so far that looks good to me too. Thanks.

**Comment by R101G** [comment]
@rern 
When switching between RP streams I sometimes still get the problem where the stream plays and it switches to the playback page, but the metadata and cover art is not picked up until the playback page is refreshed.

**Reply by rern** [reply]
@R101G,

Please try the UPDATE again.

**Reply by R101G** [reply]
@rern 
It seemed OK for a few stream changes, then the problem re-appeared.

RP made some changes recently which affected their own app for a while, so I wonder if that is the cause.

**Comment by R101G** [comment]
@rern 

Since the update, if I select an album with replace and play I get asked to confirm that I want to replace the playlist.

**Reply by rern** [reply]
Settings > Playlist > Confirm on Replace

**Reply by R101G** [reply]
@rern 

OK, I hadn't noticed that setting! It seems a bit redundant, though, if you have already set the replace and play option.

**Comment by Garrido33** [comment]
Same here. A new update doesn't solve the problem.

**Reply by Garrido33** [reply]
Thanks Rern. Problem solved.

**Comment by rern** [comment]
@R101G, @klassikmann,

Please try the UPDATE again.

**Reply by R101G** [reply]
@rern 
I've now applied the latest update, but I still have the occasional problem of an RP channel changing, but not picking up the new metadata until the track changes or the playback screen is refreshed. I cannot see any pattern to this, so I wonder if my Pi3 is not quite fast enough sometimes.

Changing webradio streams is not so slick as it used to be. The new stream goes to play, then stop, then play, then displays default cover art, then displays the previous cover art, then the new cover art, then settles down.

rAudio 20231001
6.1.54-2 aarch64
Raspberry Pi 3B Rev 1.2
BCM2837 • 956 MiB
4 x Cortex-A53 @ 1.20 GHz

**Comment by R101G** [comment]
@rern 
I still have this problem of the metadata not always being picked up when changing streams, particularly with RP. I think this is only happening when I use the Android tablet, both with the app or just using the browser. Firefox on a desktop seems OK.

My tablet is a cheap one, and is only used to control rAudio, so is it possible that it is a bit too slow sometimes? If so, is there any way to allow for this?

**Reply by rern** [reply]
Is it better with update 20231210 ?

**Reply by R101G** [reply]
@rern 
I still sometimes have to refresh the page after selecting an RP station - I would say about 25% of the time. Sometimes it gets the data for a second, and then loses it until the next track change.

**Comment by R101G** [comment]
@rern 
It might be a separate problem, but 20231210 update seems to have lost cover art for any webradio except the RP streams.

**Reply by cmh714** [reply]
Not that it helps but working here.

**Reply by R101G** [reply]
I wonder if this is the same problem as #1516. I tried an earlier version, which worked until I updated it.

**Comment by R101G** [comment]
Closing this thread as I have reverted to 20231118 update as a workaround.
